cloud/gcp: TestAntagonisticGCSRead failed

cloud/gcp.TestAntagonisticGCSRead [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/11499199?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/11499199?buildTab=artifacts#/) on release-23.1.9-rc @ [49148609ccafb4dd3bc547ef3aaa10c95b4f8dbf](https://github.com/cockroachdb/cockroach/commits/49148609ccafb4dd3bc547ef3aaa10c95b4f8dbf):


```
        	            				github.com/cockroachdb/cockroach/pkg/cloud/gcp/gcs_storage_test.go:331
        	Error:      	Received unexpected error:
        	            	open antagonistic-read/test-antagonistic-read: storage: object doesn't exist: gcs object "antagonistic-read/test-antagonistic-read" does not exist: external_storage: file doesn't exist
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/cloud/gcp.(*gcsStorage).ReadFileAt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cloud/gcp/gcs_storage.go:307
        	            	  | github.com/cockroachdb/cockroach/pkg/cloud/gcp.(*gcsStorage).ReadFile
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cloud/gcp/gcs_storage.go:276
        	            	  | github.com/cockroachdb/cockroach/pkg/cloud.(*esWrapper).ReadFile
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cloud/impl_registry.go:331
        	            	  | github.com/cockroachdb/cockroach/pkg/cloud/cloudtestutils.CheckAntagonisticRead
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cloud/cloudtestutils/cloud_test_helpers.go:545
        	            	  | github.com/cockroachdb/cockroach/pkg/cloud/gcp.TestAntagonisticGCSRead
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cloud/gcp/gcs_storage_test.go:331
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1446
        	            	Wraps: (2) open antagonistic-read/test-antagonistic-read: storage: object doesn't exist
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/cloud/gcp.(*gcsStorage).ReadFileAt
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cloud/gcp/gcs_storage.go:308
        	            	  | github.com/cockroachdb/cockroach/pkg/cloud/gcp.(*gcsStorage).ReadFile
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cloud/gcp/gcs_storage.go:276
        	            	  | github.com/cockroachdb/cockroach/pkg/cloud.(*esWrapper).ReadFile
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cloud/impl_registry.go:331
        	            	  | github.com/cockroachdb/cockroach/pkg/cloud/cloudtestutils.CheckAntagonisticRead
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cloud/cloudtestutils/cloud_test_helpers.go:545
        	            	  | github.com/cockroachdb/cockroach/pkg/cloud/gcp.TestAntagonisticGCSRead
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cloud/gcp/gcs_storage_test.go:331
        	            	  | [...repeated from below...]
        	            	Wraps: (4) gcs object "antagonistic-read/test-antagonistic-read" does not exist
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/cloud.init
        	            	  | 	github.com/cockroachdb/cockroach/pkg/cloud/external_storage.go:118
        	            	  | runtime.doInit
        	            	  | 	GOROOT/src/runtime/proc.go:6349
        	            	  | runtime.doInit
        	            	  | 	GOROOT/src/runtime/proc.go:6326
        	            	  | runtime.doInit
        	            	  | 	GOROOT/src/runtime/proc.go:6326
        	            	  | runtime.main
        	            	  | 	GOROOT/src/runtime/proc.go:233
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	Wraps: (6) external_storage: file doesn't exist
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.leafError
        	Test:       	TestAntagonisticGCSRead
--- FAIL: TestAntagonisticGCSRead (3.10s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAntagonisticGCSRead.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30974

--- a/pkg/cloud/cloudtestutils/cloud_test_helpers.go
+++ b/pkg/cloud/cloudtestutils/cloud_test_helpers.go
@@ -510,7 +510,7 @@ func CheckAntagonisticRead(
 ) {
 	rnd, _ := randutil.NewTestRand()
 
-	const basename = "test-antagonistic-read"
+	basename := fmt.Sprintf("test-antagonistic-read-%d", NewTestID())
 	data, cleanup := uploadData(t, testSettings, rnd, conf, basename)
 	defer cleanup()
 

